src/main/kotlin/tools/aqua/konstraints/dsl/SMTFunction.kt

```diff
@@ -233,9 +233,15 @@ fun <T : Sort, S1 : Sort, S2 : Sort, S3 : Sort, S4 : Sort, S5 : Sort> SMTProgram
  */
 class Const<T : Sort>(val sort: T, val program: SMTProgramBuilder, val name: String) {
   operator fun getValue(thisRef: Any?, property: KProperty<*>): Expression<T> {
-    program.registerFun(name, sort, emptyList())
-    return UserDeclaredExpression(name.symbol(), sort)
+    if (const == null) {
+      program.registerFun(name, sort, emptyList())
+      const = UserDeclaredExpression(name.symbol(), sort)
+    }
+
+    return checkNotNull(const)
```

Comment:
This is thread-unsafe. I don't think that parallel construction is likely to happen often, but we could avoid this by either eagerly initializing `const` or using double-checked locking.

Reply:
this is repeated for every `by`-able type